crates/host/src/nats/store.rs

```diff
@@ -0,0 +1,299 @@
+//! Implementation of the [crate::config::ConfigManager] and [crate::store::StoreManager] traits for NATS JetStream KV [Store].
+
+use std::{collections::HashMap, sync::Arc};
+
+use anyhow::{anyhow, ensure, Context as _};
+use async_nats::jetstream::kv::{Entry as KvEntry, Operation, Store};
+use bytes::Bytes;
+use futures::{StreamExt as _, TryStreamExt as _};
+use tokio::{
+    sync::watch::{self, Receiver},
+    task::JoinSet,
+};
+use tracing::{debug, error, instrument, warn};
+
+use crate::{
+    config::ConfigManager,
+    store::StoreManager,
+    wasmbus::{
+        claims::{Claims, StoredClaims},
+        ComponentSpecification,
+    },
+};
+
+#[async_trait::async_trait]
+impl StoreManager for Store {
+    #[instrument(level = "debug", skip(self))]
+    async fn get(&self, key: &str) -> anyhow::Result<Option<Bytes>> {
+        self.get(key)
+            .await
+            .map_err(|err| anyhow::anyhow!("Failed to get config: {}", err))
+    }
+
+    #[instrument(level = "debug", skip(self, value))]
+    async fn put(&self, key: &str, value: Bytes) -> anyhow::Result<()> {
+        self.put(key, value)
```

Comment:
Are there any worries about race conditions here? In other words, should we be using the revision number?